--- scripts/package/builddeb_org	2018-08-11 16:14:12.776543219 +0200
+++ scripts/package/builddeb	2018-08-11 16:20:42.455408047 +0200
@@ -81,7 +81,7 @@
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
 fi
-cp "$($MAKE -s image_name)" "$tmpdir/$installed_image_path"
+cp "arch/arm/boot/uImage" "$tmpdir/boot/uImage"

 if grep -q "^CONFIG_OF=y" $KCONFIG_CONFIG ; then
 	# Only some architectures with OF support have this target
